### DISCUSSION TITLE
HACS vs Github install help

### DISCUSSION THREAD

**Discussion by sdi72** [ExperienceLovelace/ha-floorplan#77] [Q&A] [answered]
Thanks Petar for recommending I ask here.

I am completely stuck on how to complete the HA-floorplan process. I've spent alot of time learning SweetHome3D and I've watched every video and read every quick start I can find.

I've installed ha-floorplan through hacks and verified the resource was added to hacsfiles
I've created the new view and can see the example csv
There were no css or svg files so i created the floorplan directory and uploaded the files
File manager shows:
www/community/ha-floorplan for floorplan.js and floorplan.js.gz
www/floorplan/examples/home/ for home.css and home.svg

I cannot replace the example home.svg with my own plan. I deleted home.svg and restarted HA. It still showed the example. Then uploaded my own plan and it still showed the example.

**Comment by exetico** [comment, marked as answer]
Hi sid72,

To me it sounds like you're facing problems with the caching-function. Please reload the page without caching of the SVG-element.

See this if you'll try to disable caching:
https://github.com/ExperienceLovelace/ha-floorplan/discussions/58#discussioncomment-501477

**Reply by sdi72** [reply]
Thank you. I am very new to HA and programming. Do I just add cashe=false? Where, in the home.css anywhere it points to the floorplan folder?

**Reply by exetico** [reply]
It should be placed in your configuration (YAML-code). Please follow the link, and check the example.

**Reply by sdi72** [reply]
ok that may be the problem. I haven't added any floorplan code in config.yaml

**Reply by exetico** [reply]
Maybe you should follow the step by step guide, we have in the docs, and after that, follow the links in the step-by-step guide, to the last steps. There's everything explained, almost as much as possible 😊

**Reply by exetico** [reply]
https://experiencelovelace.github.io/ha-floorplan/docs/quick-start/
+
https://community.home-assistant.io/t/floorplan-now-available-as-a-lovelace-card/115489/323?u=exetico

**Reply by sdi72** [reply]
I really do appreciate that. I did follow the quickstart. I am looking at the same sample floorplan as step 13 and in no steps did we edit a config.yaml. Step 5 has you copy code to the view and thats what I did.

**Reply by exetico** [reply]
Did it work?

**Reply by sdi72** [reply]
Thank you for asking but I don't know what to do. Like I said above, I followed the quickstart steps, I can see the sample plan in step 13, but don't know where to add code for cache. Nowhere in the quickstart did we edit config.yaml unless I missed something, which is highly likeley.

**Reply by exetico** [reply]
I'll suggest you to share some screenshots, if your still unsure about what to do. We can't help, without more insight.

I'd also start over, if i was facing the same issues, just to follow the steps one by one, and see how it goes.

You can **still** check my first link, to see there the `cache` option goes.

If yours unsure, you're also welcome to share your YAML configs.

Do something, so we're able to give more inputs. I personally can't help without more details about your current setup including copy of your YAML and more, if that's relevant, too.

**Reply by sdi72** [reply]
OK. I've uninstalled HA floorplan through HACS and removed floorplan folders from config. My config/www only has a community folder which is empty.

I reinstalled HA-Floorplan through HACS
www/community/ha-floorplan/floorplan.js and floorplan.gz are reinstalled

I went to config, resources and added:
url: /hacsfiles/ha-floorplan/floorplan.js
type: module

I created view , went to raw config editor and pasted code

I can see a blank floorplan card, step 6

I went to file editor and in www added floorplan/examples/home

THIS TIME I only uploaded my custom home.svg, then went to the github and uploaded the css from hafloorplanmaster/docs/docs/floorplan

Go to my lovelace view and now have a big red box that says floorplan-card doesnt exist

**Reply by sdi72** [reply]
this is all I have in config.yaml


# Configure a default setup of Home Assistant (frontend, api, etc)
default_config:

# Text to speech
tts:
  - platform: google_translate

group: !include groups.yaml
automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml

influxdb:
  host: 
  port: 
  database: homeassistant
  username: homeassistant
  password: 
  max_retries: 3
  default_measurement: state

**Reply by sdi72** [reply]
I've deleted my entire VM and started over with HA. Followed these steps again. I've installed through HACS, added view and pasted code, added folders and uploaded home files,

Do not see sample floorplan and only have a red log level box that says:
Custom element doesn't exist: floorplan-card

What specifically resolves this?

**Reply by OzGav** [reply]
Are you coding up your ui-lovelace.yaml in YAML mode (ie directly editing with a text editor) or via the interface in home assistant? Either way have you loaded the floorplan.js resource? In YAML mode you should have this in your configuration.yaml
```
lovelace:
  mode: yaml
  # Include external resources only add when mode is yaml, otherwise manage in the resources in the lovelace configuration panel.
  resources:
  - type: module
    url: /hacsfiles/ha-floorplan/floorplan.js
```
in UI mode you should have added it through CONFIGURATION>>LOVELACE DASHBOARDS>>RESOURCES

**Reply by sdi72** [reply]
File editor yaml. Coincidentally I just installed a custom badge card from HACS and added this at the top which also fixed floorplan!

type: 'custom:badge-card'
views:
  - path: default_view
    title: Home
    cards:
      - type: 'custom:badge-card'
        badges:
          - entity: person.
          - entity: person.
          - entity: person.
          - entity: person.
          - entity: sun.sun

However the floorplan is only using 1/6 of the view space. How do I make it take the full page? Can I still add that lovelace yaml code under the floorplan section? Sorry I know these are beginner questions

**Reply by OzGav** [reply]
Use the` panel: true` option
```
title: Home Assistant
views:
  - panel: true
    cards:
      - config:
          image:
            location: /local/floorplan/floorplan_3d-light.svg?v=1
            cache: true

...
```

**Reply by OzGav** [reply]
Also when you post in code can you then highlight it and then press the <> symbol so we can see the code more easily.

**Reply by sdi72** [reply]
Thank you Sir. 
I was incorrect. I'm editing lovelace yaml through raw editor and other files through file editor. I always had the resource loaded in config/lovelace/resources. There was never anything added to config.yaml pertaining to floorplan and I did not see that in the quickstart.
I just added the lovelace code above into config.yaml, to point to floorplan.js. I rebooted and now have a fresh overview like a new install, and the floorplan view is gone. I do have a snapshot I can go back to.
I know I am a few years behind everyone on the learning curve sorry

**Reply by sdi72** [reply]
Well, now that I went back to the snapshot, the lovelace code is gone from config.yaml but it is still in yaml mode and I cannot edit anything through the UI.

**Reply by OzGav** [reply]
Not sure why you can't edit in the UI if there is no reference to lovelace: in the configuration.yaml 

Either way you can add this to configuration.yaml and it should allow you back into the UI editor

```
lovelace:
  mode: storage
```

**Reply by sdi72** [reply]
Thank you. Learning something new every day

**Reply by exetico** [reply]
So, is it fixed, or?

**Reply by sdi72** [reply]
yes thank you all, I can see floorplan all that's left is config questions

**Reply by exetico** [reply]
Please select the answer to the question (this thread), and if you have any config-specific question, please create a separate question for that 😊

**Comment by cwoodhouse** [comment]
Sorry for the newbie question.  What does this mean in the Quickstart?  Where is the so-called "resource file"?

"Remember to add the /hacsfiles/ha-floorplan/floorplan.js as a module to your resource file - like you do with other modules."

**Reply by OzGav** [reply]
> Sorry for the newbie question. What does this mean in the Quickstart? Where is the so-called "resource file"?

https://github.com/ExperienceLovelace/ha-floorplan/discussions/77#discussioncomment-550519

**Reply by exetico** [reply]
And that's only relevant, if you're in YAML-mode. Else, it's handled in the GUI.

**Reply by OzGav** [reply]
Hi @cwoodhouse just checking if the problem has been solved. If so it would be great if you could mark the reply that answered your question.

**Reply by cwoodhouse** [reply]
Yes, thanks!  I found the relevant resources and I got started on my floorplan.
